src/v2/Apps/Show/__tests__/ShowInstallShots.jest.tsx

```diff
@@ -13,62 +13,47 @@ jest.mock("@artsy/palette", () => {
   }
 })
 
-const SHOW_FIXTURE = {
-  name: "Example Show",
-  images: [
-    {
-      internalID: "example-image-0",
-      mobile: { width: 449, height: 300 },
-      desktop: {
-        src: "0_example1x.jpg",
-        srcSet: "0_example1x.jpg",
-        width: 599,
-        height: 400,
-      },
-      zoom: {
-        src: "0_zoomExample1x.jpg",
-        srcSet: "0_zoomExample1x.jpg",
-        width: 900,
-        height: 600,
-      },
-    },
-    {
-      internalID: "example-image-1",
-      mobile: { width: 449, height: 300 },
-      desktop: {
-        src: "1_example1x.jpg",
-        srcSet: "1_example1x.jpg",
-        width: 599,
-        height: 400,
-      },
-      zoom: {
-        src: "1_zoomExample1x.jpg",
-        srcSet: "1_zoomExample1x.jpg",
-        width: 900,
-        height: 600,
-      },
-    },
-  ],
-}
+jest.unmock("react-relay")
+
+const { getWrapper } = setupTestWrapper({
+  Component: ShowInstallShotsFragmentContainer,
+  query: graphql`
+    query ShowInstallShots_Test_Query {
+      show(id: "xxx") {
+        ...ShowInstallShots_show
+      }
+    }
+  `,
+})
 
 describe("ShowInstallShots", () => {
-  const getWrapper = (data = SHOW_FIXTURE) => {
-    return mount(<ShowInstallShots show={data as any} />)
-  }
+  it("renders empty string if there are no images", () => {
+    const wrapper = getWrapper({ Show: () => ({ images: [] }) })
 
-  it("renders null if there are no images", () => {
-    const wrapper = getWrapper({ ...SHOW_FIXTURE, images: [] })
-    expect(wrapper.html()).toBeNull()
+    expect(wrapper.html()).toBe("")
   })
 
   it("renders the images", () => {
-    const wrapper = getWrapper()
+    const wrapper = getWrapper({
+      Show: (_, generateID) => ({
+        images: [{ internalID: generateID() }, { internalID: generateID() }],
+      }),
+    })
+
     expect(wrapper.find("button")).toHaveLength(2)
     expect(wrapper.find("img")).toHaveLength(2)
   })
 
   it("zooms the second image when it is clicked", () => {
-    const wrapper = getWrapper()
+    const wrapper = getWrapper({
+      Show: (_, generateID) => ({
+        images: [
+          { internalID: generateID() },
+          { internalID: generateID(), zoom: { src: "1_zoomExample1x.jpg" } },
+        ],
```

Comment:
I'm a little unclear as to whether or not there's a nicer way to do this: controlling the number of mocks generated when you have array fields.